soc/xtensa/esp32/Kconfig.defconfig

```diff
@@ -29,6 +29,14 @@ config SOC_TOOLCHAIN_NAME
 	string
 	default "espressif_esp32"
 
+# Both options below previously leaking from intel_adsp, not directly used by
+# esp32 but required for building
+config XTENSA_CACHED_REGION
```

Comment:
where this is used by esp32 buildings? I'm saying that because, AFAIK, portions of the startup code and the crt code from xtensa arch is not used by esp32 since it is handled by its rom code.

Reply:
@uLipe, both this kconfig values are needed for arch/xtensa, even not used by esp32. This shall be handled in a new PR.

Reply:
fair enough

Reply:
Right, xtensa doesn't really have a single/unified boot path right now; everyone does it differently (something that needs to be fixed -- we inherited a crt.S implementation from Cadence that didn't really work for anyone).

This particular trick is specifically for intel_adsp devices, which map memory twice in two different Region Protection Option regions.  But beyond the double-mapping, it's all implemented using documented Cadence/Xtensa interfaces, so it lives in the arch layer.